skimage/_shared/filters.py

```diff
@@ -134,8 +171,8 @@ def gaussian(
     image = convert_to_float(image, preserve_range)
     float_dtype = _supported_float_type(image.dtype)
     image = image.astype(float_dtype, copy=False)
-    if (output is not None) and (not np.issubdtype(output.dtype, np.floating)):
-        raise ValueError("Provided output data type is not float")
+    if (out is not None) and (not np.issubdtype(out.dtype, np.floating)):
+        raise ValueError("Provided out data type is not float")
```

Comment:
```suggestion
        raise ValueError("Provided `out` data type is not float")
```
Works better for the user reading their screen.